lightly/api/api_workflow_download_dataset.py

```diff
@@ -269,9 +274,10 @@ def export_filenames_by_tag_id(
             A list of the samples filenames within a certain tag.
 
         """
-        filenames = self._tags_api.export_tag_to_basic_filenames(
-            self.dataset_id,
-            tag_id,
+        filenames = retry(
```

Comment:
That retry is not necessary.

Reply:
why shouldn't it be necessary?

Reply:
I would remove the retry for two reasons:
- a) Consistency with all the other endpoints.
- b) Some of the endpoints are already retried by the docker. Thus double retries can happen, which can make the solution quickly unresponsive.

Reply:
I don't think "making it consistent" is the correct approach when the consistency is just plain wrong/stupid. Retrying things is good as the http request can fail for multiple reasons of which we can't influence many.
I don't see how a double retry can happen unless someone wraps an api_workflow_client function in a retry. Looking at the docker code, I don't see any place where that happens.

I don't know what the pythoneers plan is to address this issue or why you all are so totally against retrying stuff but just letting processes die after a couple of hours 'just because' is not a solution.

So you can choose
1. I will leave it this way
2. I will remove it as you want and also remove any retry logic of the pagination helper

Reply:
@japrescott You have convinced me, let's keep the retry logic here.